README.md

```diff
@@ -41,6 +41,74 @@ To compile all source files once:
 
 ## Details and Differences
 
+### Classes
+
+Creating a file class in GDScript requires TypeScript class to be marked with `export default` keywords.
+
+```ts
+export default class MyNode extends Node2D {
+  constructor() {
+    super()
+    print("Hello world")
+  }
+}
+```
+
+```gdscript
+extends Node2D
+class_name MyNode
+
+func _ready():
+  print("Hello world")
+```
+
+File classes can also be anonymous to skip `class_name` generation.
+
+```ts
+export default class extends Node2D {
+  constructor() {
+    super()
+    print("Hello world")
+  }
+}
+```
+
+```gdscript
+extends Node2D
+
+func _ready():
+  print("Hello world")
+```
+
+### Inner classes
+
+To generate inner classes in GDScript file, TypeScript classes should not be marked with `default` keyword.
+
+```ts
+export class InnerClass extends Node2D {
+  constructor() {
+    super()
+    print("Hello InnerClass")
+  }
+}
+
+class NotExportedInnerClass {
+  constructor() {
+    print("Hello NotExportedInnerClass")
+  }
+}
+```
+
+```gdscript
+class InnerClass extends Node2D:
+  func _init().():
```

Comment:
The .() at the end seems incorrect.

Reply:
This means that a base constructor is called ([here are docs](https://docs.godotengine.org/en/stable/getting_started/scripting/gdscript/gdscript_basics.html#class-constructor)). If base constructor does not have arguments `.()` can be omitted. Why is it wrong? Does this code not compile or is wrongly generated?

Reply:
Yeah I know it looks weird but it really is valid GD syntax!